travharv/execution_report.py

```diff
@@ -0,0 +1,207 @@
+        """
+        log.debug(
+            "Reporting the results of the task execution report to the store"
+        )
+        assertion_report_id = assertion_report.id
+        last_mod = assertion_report.assertion_time
+
+        log.debug(f"Assertion Report ID: {assertion_report_id}")
+        log.debug(f"Last Modified: {last_mod}")
+
+        self.assertion_reports.append(assertion_report)
+        self.report_content["last_mod"] = last_mod
+
+
+class ExecutionReport:
+    """
+    A class to represent an ExecutionReport.
+    This class will report the results of the assertions per task per config.
+    e.g.: All TaskExecutionReports will be stored in a dictionary
+    """
+
+    def __init__(self, rdf_store_access: RDFStoreAccess, config_name: str):
+        """constructor
+
+        :param rdf_store_access: RDFStoreAccess
+        """
+        self.rdf_store_access = rdf_store_access
+        log.debug("ExecutionReport initialized")
+        self.report_content = {
+            "last_mod": timestamp(),
+            "config_name": config_name,
+            "id": uuid4(),
+        }
+        self.task_reports = []
+
+    def add_task_report(self, task_execution_report: TaskExecutionReport):
+        """
+        Report the results of the assertions to the store.
+        """
+        log.debug("Reporting the results of the execution report to the store")
+        # get info needed for update
+        task_id = task_execution_report.report_content["task_id"]
+        last_mod = task_execution_report.report_content["last_mod"]
+
+        log.debug(f"Task ID: {task_id}")
+        log.debug(f"Last Modified: {last_mod}")
+
+        self.task_reports.append(task_execution_report)
+        # TODO: write template for reporting triples
+        # to store with the report_content
+
+        log.debug(f"{task_execution_report.report_content['task_id']=}")
+        for assertion in task_execution_report.assertion_reports:
+            log.debug(f"{assertion.id=}")
+            log.debug(f"{assertion.assertion_path=}")
+            log.debug(f"{assertion.assertion_result=}")
+            log.debug(f"{assertion.message=}")
+            log.debug(f"Graphs added: {len(assertion.graph_reports)}")
+            # count up the ammount of triples that are
+            # located in assertion.graph_reports[i]["triples"]
+            c_triples = 0
+            for graph in assertion.graph_reports:
+                c_triples += graph.triple_count
+            log.debug(f"Triples added: {c_triples}")
+
+        self._make_report_graph()
+
+    def _make_report_graph(self):
+        pre_ttl = QUERY_BUILDER.build_syntax(
+            "execution_report.ttl",
+            execution_report=self.report_content,
+            task_reports=self.task_reports,
+        )
+
+        log.debug(f"{pre_ttl}")
+
+        self.execution_report_graph = Graph()
+        self.execution_report_graph.parse(data=pre_ttl, format="turtle")
+
+    def report_to_store(self):
```

Comment:
this becomes the only point where rdf_store is needed in this new setup?

it feels like it can be removed as member from this class (and thus also the __init__ call) and can just be handed as argument in this call?

point being "the store" does not feel to be a part of the "state" of the report instance, only the actually gathered report-data makes up that state imho